# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/front/spv/mod.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/naga/src/front/spv/mod.rs
--- a/third_party/rust/naga/src/front/spv/mod.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/naga/src/front/spv/mod.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -82,6 +82,8 @@
     spirv::Capability::GroupNonUniformBallot,
     spirv::Capability::GroupNonUniformShuffle,
     spirv::Capability::GroupNonUniformShuffleRelative,
+    spirv::Capability::RuntimeDescriptorArray,
+    spirv::Capability::StorageImageMultisample,
     // tricky ones
     spirv::Capability::UniformBufferArrayDynamicIndexing,
     spirv::Capability::StorageBufferArrayDynamicIndexing,
@@ -90,6 +92,7 @@
     "SPV_KHR_storage_buffer_storage_class",
     "SPV_KHR_vulkan_memory_model",
     "SPV_KHR_multiview",
+    "SPV_EXT_descriptor_indexing",
     "SPV_EXT_shader_atomic_float_add",
     "SPV_KHR_16bit_storage",
 ];
@@ -2799,6 +2802,7 @@
                     let options = image::SamplingOptions {
                         compare: false,
                         project: false,
+                        gather: false,
                     };
                     self.parse_image_sample(
                         extra,
@@ -2815,6 +2819,7 @@
                     let options = image::SamplingOptions {
                         compare: false,
                         project: true,
+                        gather: false,
                     };
                     self.parse_image_sample(
                         extra,
@@ -2831,6 +2836,7 @@
                     let options = image::SamplingOptions {
                         compare: true,
                         project: false,
+                        gather: false,
                     };
                     self.parse_image_sample(
                         extra,
@@ -2847,6 +2853,41 @@
                     let options = image::SamplingOptions {
                         compare: true,
                         project: true,
+                        gather: false,
+                    };
+                    self.parse_image_sample(
+                        extra,
+                        options,
+                        ctx,
+                        &mut emitter,
+                        &mut block,
+                        block_id,
+                        body_idx,
+                    )?;
+                }
+                Op::ImageGather => {
+                    let extra = inst.expect_at_least(6)?;
+                    let options = image::SamplingOptions {
+                        compare: false,
+                        project: false,
+                        gather: true,
+                    };
+                    self.parse_image_sample(
+                        extra,
+                        options,
+                        ctx,
+                        &mut emitter,
+                        &mut block,
+                        block_id,
+                        body_idx,
+                    )?;
+                }
+                Op::ImageDrefGather => {
+                    let extra = inst.expect_at_least(6)?;
+                    let options = image::SamplingOptions {
+                        compare: true,
+                        project: false,
+                        gather: true,
                     };
                     self.parse_image_sample(
                         extra,
@@ -3869,7 +3910,10 @@
                             crate::Barrier::TEXTURE,
                             semantics & spirv::MemorySemantics::IMAGE_MEMORY.bits() != 0,
                         );
+
+                        block.extend(emitter.finish(ctx.expressions));
                         block.push(crate::Statement::ControlBarrier(flags), span);
+                        emitter.start(ctx.expressions);
                     } else {
                         log::warn!("Unsupported barrier execution scope: {exec_scope}");
                     }
@@ -3911,7 +3955,10 @@
                         crate::Barrier::TEXTURE,
                         semantics & spirv::MemorySemantics::IMAGE_MEMORY.bits() != 0,
                     );
+
+                    block.extend(emitter.finish(ctx.expressions));
                     block.push(crate::Statement::MemoryBarrier(flags), span);
+                    emitter.start(ctx.expressions);
                 }
                 Op::CopyObject => {
                     inst.expect(4)?;